### ISSUE TITLE
Mute certain XHR requests which are convoluting test log on UI

### ISSUE THREAD

**Issue by szerintedmi** [cypress-io/cypress#1184]
- Operating System: MacOs
- Cypress Version: 1.4.1
- Browser Version: Chrome 63 & Electron 53
- React: 16.2.0
- web3: 1.0.0-beta.27

### Is this a Feature or Bug?
Feature / question

### Current behavior:
E2E testing dapp using web3. The app uses web3 which is polling a port regularly. This filling up test run logs on cypress UI. XHR events are all around while running and it's logging even after successful finish it makes really hard to follow tests. It also makes the UI very unresponsive.

### Desired behavior:
Would be great to silence / mute certain XHR calls in the log.
There might a way (e.g. overriding event? but I couldn't figure out..)

### How to reproduce:
Run any dapp in cypress then the logs will be filled 

#### Test code:
N/A

### Additional Info (images, stack traces, etc)
![image](https://user-images.githubusercontent.com/7456451/34823445-4433269c-f6c2-11e7-9c4f-bf6709c7a262.png)

![image](https://user-images.githubusercontent.com/7456451/34823375-0920cfaa-f6c2-11e7-862b-3ed1ee01d17d.png)

**Comment by brian-mann** [comment]
You can already do this.

It's documented here: https://docs.cypress.io/api/commands/server.html#Syntax

And to change the global defaults you can do it here: https://docs.cypress.io/api/cypress-api/cypress-server.html#Notes

**Comment by szerintedmi** [comment]
thank you, it works like a charm :)

**Comment by moinuddin14** [comment]
@szerintedmi Can you please let me know a sample code of how did you silence the XHR Request on the log. I am not running the API but UI Tests, so not sure what to do in this case

**Comment by emperol2** [comment]
@moinuddin14 Did you try to place the code below in cypress/support/index.js

`Cypress.Server.defaults({
  delay: 500,
  force404: false,
  whitelist: (xhr) => {
    // handle custom logic for whitelisting
    return true;
  }
})`

**Comment by jregen** [comment]
i don't get it.   What exactly do you put for whitelist: (xhr) => { }  ??

**Comment by miklosme** [comment]
Whitelisting all the XHR requests will break the stubbing feature, because a whitelisted request will be invisible to the `Cypress.Server`. You probably don't want to do that. See #7362.

**Comment by anil5686** [comment]
How can we do this in cypress 5.1.0
Please help

**Comment by anil5686** [comment]
Cypress.Server.defaults({
  delay: 500,
  force404: false,
  whitelist: (xhr) => {
    // handle custom logic for whitelisting
    return true;
  }
})

this code doesn't work in Cypress 5.1.0
and the using ignore in the same place is throwing an error.

**Comment by NonnaD** [comment]
> Cypress.Server.defaults({
> delay: 500,
> force404: false,
> whitelist: (xhr) => {
> // handle custom logic for whitelisting
> return true;
> }
> })
> 
> this code doesn't work in Cypress 5.1.0
> and the using ignore in the same place is throwing an error.


5.0.0 | Renamed whitelist option to ignore
 

try this 

Cypress.Server.defaults({
    delay: 500,
    force404: false,
    ignore: (xhr) => {
        // handle custom logic for whitelisting
        return true;
    }
})

**Comment by anil5686** [comment]
It's working. Thank you.

**Comment by AzrizHaziq** [comment]
I'm using `Cypress@6`, is there a way to mute with `cy.intercept`?

A suggestion add `muted: boolean` to `StaticResponse interface`

```
{
  muted?: Boolean   // default: false
  fixture?: string
  headers?: { [key: string]: string }
  statusCode?: number
  forceNetworkError?: boolean
  delayMs?: number
  throttleKbps?: number
}

**Comment by Bielik20** [comment]
Exactly the same question as AzrizHaziq. I am new to cypress and it is hard to find any information about it.

**Comment by mesqueeb** [comment]
is there any updated info on this for cypress 6?

**Comment by ArCiGo** [comment]
For me, it worked the solution of @NonnaD (I'm using Cypress v. 6.8.0 and I'm creating UI tests), but I'm not totally convinced of this workaround

**Comment by mesqueeb** [comment]
@brian-mann how to best whitelist firebase related XHR requests for Cypress 6+ ? Ideally a solution that won't be removed next version like the Server workaround I find here...

**Comment by jennifer-shehane** [comment]
Please open a new issue if you'd like to see a way to allow and/or block certain requests in cy.intercept() explaining in detail what the feature will be. Closing as this is related to the older cy.route() and the intent is a bit lost in the old closed issue.

**Comment by JessicaSachs** [comment]
Here is a new issue which is relevant for the newer `cy.intercept` commands https://github.com/cypress-io/cypress/issues/17815
